service/api_logic/user_logic.py

```diff
@@ -92,34 +103,16 @@ def confirm_token(self, token: str, expiration=3600):
         return OutputUser().dump(user)
 
 
-    async def log_in(self, email_or_username: str, password: str):
-        user = self._user_dal.get_user_by_email_or_username(email_or_username, email_or_username)
-
-        if not user:
-            self._logger.warning("User does not exist")
-            raise IncorrectUsernameOrEmailError()
-
-        if not bcrypt.checkpw(password.encode('utf-8'), user.password_hash.encode('utf-8')):
-            self._logger.warning("Passwords do not match")
-            raise IncorrectPasswordError()
-
-        token = await self.__generate_auth_token(user)
-
-        return OutputLogin(email = user.email, token = token, id = user.user_id)
+    async def log_in(self, method: str, credentials: Union[InputUserLoginDTO, InputUserByGoogleDTO]):
+        if method not in self.strategies:
+            raise ValueError("Invalid login method")
```

Comment:
Don't forget to specifically catch this exception later, so that you don't accidentally return status code 500 on API